pkg/solana/chainwriter/transform_registry.go

```diff
@@ -47,40 +56,71 @@ func CCIPExecuteArgsTransform(ctx context.Context, args any, accounts solana.Acc
 		return argsTransformed, accounts, nil
 	}
 
-	tokenPoolAddresses := []solana.PublicKey{}
-	for _, table := range registryTables {
-		tokenPoolAddresses = append(tokenPoolAddresses, table[0].PublicKey)
+	// Fetch all of the accounts in the pool lookup table with the proper IsWritable flag set
+	poolLookupAccounts, err := fetchPoolLookupAccounts(ctx, client, registryTables)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to fetch pool lookup accounts and set wrtiable flags: %w", err)
 	}
 
-	tokenIndexes := []uint8{}
-	for i, account := range accounts {
-		for _, address := range tokenPoolAddresses {
-			if account.PublicKey == address {
-				if i > 255 {
-					return nil, nil, fmt.Errorf("index %d out of range for uint8", i)
-				}
-				tokenIndexes = append(tokenIndexes, uint8(i)) //nolint:gosec
-			}
-		}
+	var aggregatedMessages []ccipocr3.Message
+	for _, report := range argsTransformed.Info.AbstractReports {
+		aggregatedMessages = append(aggregatedMessages, report.Messages...)
```

Comment:
I believe we can assume there is only going to be one report and one message, but I guess it's fine to have this for flexibility

Reply:
Ya did this in case we ever enable multiple messages to avoid one less thing to change on our side